ui/src/app/applications/components/utils.tsx

```diff
@@ -591,7 +591,18 @@ function getActionItems(
         concat([[] as MenuItem[]], resourceActions), // this resolves at first to [] and then whatever the API returns
         concat([[] as MenuItem[]], execAction), // this resolves at first to [] and then whatever the API returns
         concat([[] as MenuItem[]], links) // this resolves at first to [] and then whatever the API returns
-    ).pipe(map(res => ([] as MenuItem[]).concat(...res)));
+    ).pipe(
+        map(res => {
+            const combinedItems = ([] as MenuItem[]).concat(...res);
+            return combinedItems.map(
+                item =>
+                    ({
+                        ...item,
+                        disabled: item.title === 'Delete' && resource.status === appModels.SyncStatuses.OutOfSync && resource.health?.status === appModels.HealthStatuses.Missing
```

Comment:
Could this disablement logic be moved up to where the Delete action is defined?  Perhaps change what's there to ActionMenuItem?

Reply:
Is it possible to have a Missing resource with an Unknown status? I do see a Missing resource with Unsynced status.